packages/analytics-js/src/components/pluginsManager/PluginsManager.ts

```diff
@@ -87,6 +89,16 @@ class PluginsManager implements IPluginsManager {
       );
     }
 
+    // Consent manager related plugins
+    const selectedConsentManager: string | undefined = getUserSelectedConsentManager(
+      state.consents.cookieConsentOptions.value,
+    );
+    if (!selectedConsentManager || !SUPPORTED_CONSENT_MANAGERS.includes(selectedConsentManager)) {
+      pluginsToLoadFromConfig = pluginsToLoadFromConfig.filter(
+        pluginName => pluginName !== (PluginName.ConsentManager || PluginName.OneTrust),
```

Comment:
We have to support other consent managers beyond OneTrust. So, only the appropriate plugin needs to be allowed to load.